arm/curve25519/bignum_cmul_p25519.S

```diff
@@ -44,6 +47,7 @@
 
 S2N_BN_SYMBOL(bignum_cmul_p25519):
```

Comment:
I am a bit confused here. Is the idea that `bignum_cmul_p25519` and `bignum_cmul_p25519_alt` are the same function, i.e. one is an alias for the other? I wonder if this pattern is allowed by all ABIs and/or if it might be the source of some of the issues here. In OpenSSL when they have such aliases, the alias is implemented as a whole separate function that `jmp`s to the function it is aliasing.

Reply:
We were initially trying to support a uniform naming scheme for both ARM and x86. The bifurcation into "alt" and "non-alt" arises when _for at least one architecture_ there is value in different variants with different microarchitectural targets. Cases like this where the same function has both names reflect those where on the other architecture (only) there is value in two versions, but on this architecture it didn't seem worth it. Currently, consumers like AWS-LC handle the dispatch according to uarch. Now, we no longer quite maintain the uniformity of this naming scheme since there are a few extra functions for ARM, so perhaps we should revisit this.